server/datastore/mysql/apple_mdm_test.go

```diff
@@ -954,7 +955,20 @@ func configProfileForTest(t *testing.T, name, identifier, uuid string) *fleet.MD
 </dict>
 </plist>
 `, name, identifier, uuid))
-	cp, err := fleet.NewMDMAppleConfigProfile(prof, nil)
+}
+
+func configProfileForTest(t *testing.T, name, identifier, uuid string) *fleet.MDMAppleConfigProfile {
+	prof := configProfileBytesForTest(name, identifier, uuid)
+	cp, err := fleet.NewMDMAppleConfigProfile(configProfileBytesForTest(name, identifier, uuid), nil)
+	sum := md5.Sum(prof) // nolint:gosec // used only to hash for efficient comparisons
+	cp.Checksum = sum[:]
+	require.NoError(t, err)
```

Comment:
Nit: you should do the error check right after the call to `NewMDMAppleConfigProfile`, otherwise `cp` could be nil.